### PR TITLE
Replace pthreads by std::threads and unify layout

### PR DESCRIPTION
**Pull request by chris-hld**
Switching from pthreads to std::threads. With that the SSR itself is now completely on std::threads.

### CODE REVIEW

File: src/trackerpolhemus.cpp [hunk 1]
```diff
@@ -201,44 +200,42 @@ void
...
 
 void*
-ssr::TrackerPolhemus::thread(void *arg)
+ssr::TrackerPolhemus::_thread(void *arg)
 {
   char c;
   std::string line;
 
-  while (!_stopped)
+  struct pollfd fds;
```
Comment (chris-hld) [inline src/trackerpolhemus.cpp]: I'm assuming moving this out of the while loop improves performance

**Comment by chris-hld** [comment]
Has anyone access to the sources/headers so that we can at least check that it compiles?

**Comment by JensAhrens** [comment]
Which sources/headers?

**Comment by chris-hld** [comment]
Sorry, just found a link in the docs
https://ssr.readthedocs.io/en/0.5.0/operation.html#head-tracking
I'll try to compile with all head trackers enabled.

**Comment by chris-hld** [comment]
Could include and compile every supported tracker!

**Comment by mgeier** [comment]
Thanks @chris-hld, that's great!

Do you really need the thread ID?

What about using `.joinable()` to check if the thread should be joined?

https://github.com/SoundScapeRenderer/ssr/blob/badf8c00ae24360b93fc2c26d304099fce8b7f78/src/websocket/server.h#L79

Also, the crazy re-direction via `void*` is not necessary anymore, you can directly pass a member function to the `std::thread` constructor, e.g.

https://github.com/SoundScapeRenderer/ssr/blob/badf8c00ae24360b93fc2c26d304099fce8b7f78/src/websocket/server.h#L73

**Comment by chris-hld** [comment]
Thank you for the suggestions, I like these changes a lot!

**Comment by chris-hld** [comment]
I tested the razor tracker.

**Pull request merged**